llarp/router/router.cpp

```diff
@@ -222,7 +222,31 @@ namespace llarp
       return false;
 #endif
 #endif
-      _identity = RpcClient()->ObtainIdentityKey();
+      constexpr int maxTries = 5;
+      int numTries = 0;
+      while (numTries < maxTries)
+      {
+        numTries++;
+        try
+        {
+          _identity = RpcClient()->ObtainIdentityKey();
+          LogWarn("Obtained lokid identity keys");
+          break;
+        }
+        catch (const std::exception& e)
+        {
+          LogWarn(
+              "Failed attempt ",
+              numTries,
+              " of ",
+              maxTries,
+              " to get lokid identity keys because: ",
+              e.what());
+
+          if (numTries == maxTries)
+            throw;
+        }
+      }
```

Comment:
This deserves a closer look. The problem is that `lokinet` will try to immediately connect to `lokid` if configured to do so, and bails out ungracefully if it can't connect.